src/Hl7.Fhir.ElementModel/ElementNode.cs

```diff
@@ -94,6 +94,37 @@ public ElementNode Add(IStructureDefinitionSummaryProvider provider, string name
             return Add(provider, child);
         }
 
+        public ElementNode Insert (IStructureDefinitionSummaryProvider provider, ElementNode child, int index, string name = null)
+        {
+            if ( provider == null )
+                throw new ArgumentNullException(nameof(provider));
+            if ( child == null )
+                throw new ArgumentNullException(nameof(child));
+
+            var childName = name ?? child.Name;
+            if ( child.Name == null ) throw Error.Argument($"The ElementNode given should have its Name property set or the '{nameof(name)}' parameter should be given.");
+
+            var childIndices = ChildList.Select((x, i) => (child: x, index: (int?)i)).Where(kvp => kvp.child.Name == childName).Select(kvp => kvp.index).ToArray();
+            if(index < 0 || index > childIndices.Length) throw Error.Argument($"The index given is out of bounds of the array size.");
+
+            var insertAtIndex = childIndices.ElementAtOrDefault(index);
+            importChild(provider, child, name, insertAtIndex ?? childIndices.Length);
```

Comment:
Explanation on `importChild` states: "Will update the child to reflect it being a child of this element, but will not yet add the child at any position within this element."
So when *is* it positioned within this element then?

Reply:
I have not made any changes to `importChild` method, but that comment on the method made me confused too